Jellyfin.Plugin.Bookshelf/Providers/GoogleBooks/GoogleBooksProvider.cs

```diff
@@ -0,0 +1,298 @@
+
+            // google rates out of five so convert to ten
+            book.CommunityRating = bookResult.volumeInfo.averageRating * 2;
+
+            if (!string.IsNullOrEmpty(bookResult.id))
+                book.SetProviderId("GoogleBooks", bookResult.id);
+
+            return book;
+        }
+
+        // convert these characters to whitespace for better matching
+        // there are two dashes with different char codes
+        private const string Spacers = "/,.:;\\(){}[]+-_=–*";
+        private const string Remove = "\"'!`?";
+
+        private static string GetComparableName(string name)
+        {
+            name = name.ToLower();
+            name = name.Normalize(NormalizationForm.FormKD);
+
+            foreach (var pair in ReplaceEndNumerals)
+            {
+                if (name.EndsWith(pair.Key))
+                {
+                    name = name.Remove(name.IndexOf(pair.Key, StringComparison.InvariantCulture), pair.Key.Length);
+                    name += pair.Value;
+                }
+            }
+
+            var sb = new StringBuilder();
+            foreach (var c in name)
+            {
+                if (c >= 0x2B0 && c <= 0x0333)
+                {
+                    // skip char modifier and diacritics 
+                }
+                else if (Remove.IndexOf(c) > -1)
+                {
+                    // skip chars we are removing
+                }
+                else if (Spacers.IndexOf(c) > -1)
+                {
+                    sb.Append(" ");
+                }
+                else if (c == '&')
+                {
+                    sb.Append(" and ");
+                }
+                else
+                {
+                    sb.Append(c);
+                }
+            }
+            name = sb.ToString();
+            name = name.Replace("the", " ");
+            name = name.Replace(" - ", ": ");
+
+            Regex regex = new Regex(@"\s+");
+            name = regex.Replace(name, " ");
+
+            return name.Trim();
+        }
+
+        private void GetBookMetadata(BookInfo item)
+        {
+            foreach (var regex in NameMatches)
+            {
+                var match = regex.Match(item.Name);
+                if (!match.Success) continue;
+
+                // catch return value because user may want to index books from zero
+                // but zero is also the return value from int.TryParse failure
+                var result = int.TryParse(match.Groups["index"]?.Value, out var index);
+                if (result) item.IndexNumber = index;
+
+                item.Name = match.Groups["name"].Value.Trim();
+
+                // might as well catch the return value here as well
+                result = int.TryParse(match.Groups["year"]?.Value, out var year);
+                if (result) item.Year = year;
+            }
+        }
+
+        static readonly Dictionary<string, string> ReplaceEndNumerals = new Dictionary<string, string> {
```

Comment:
Add an access modifier e.g private.
Also `private static` fields should be prefixed with an underscore and be camelBack.